hs_core/models.py

```diff
@@ -1465,10 +1465,18 @@ def delete(self, using=None):
                 if fl.fed_resource_file_name_or_path.find(self.short_id) >= 0:
```

Comment:
@pkdash No, it won't. self.root_path and self.resource_federation_path are all resource level properties that only consider the resource root path and federation path which is not sufficient and does not apply the file-level granularity here. In this specific case, resource is in hydroshare zone, but the files being added into the resource comes from a federated user zone that was indicated by the fed_resource_file_name_or_path property. This can probably be simplified by clearing up fed_resource_file_name_or_path property after the file is added into the resource that resides in hydroshare zone rather than a federated zone, but I am afraid I could be breaking things up if I do so at this point, especially given @alvacouch is also doing work to simplify resource files at this time. To minimize effort on my part and maximize efficiency between my efforts and @alvacouch 's effort, I think what I did here is the simplest and safest fix to get the issue resolved without causing any potential regression issues. I also added a unit test case to cover this special scenario so that we will run into less potential regression issues during refactoring work.